hudi-integ-test/src/main/java/org/apache/hudi/integ/testsuite/dag/nodes/BaseValidateDatasetNode.java

```diff
@@ -157,14 +157,19 @@ public void execute(ExecutionContext context, int curItrCount) throws Exception
     }
   }
 
-  private Dataset<Row> getInputDf(ExecutionContext context, SparkSession session, String inputPath) {
+  private Dataset<Row> getInputDf(ExecutionContext context, SparkSession session, String inputPath, String partitionsToSkipWithValidate) {
     String recordKeyField = context.getWriterContext().getProps().getString(DataSourceWriteOptions.RECORDKEY_FIELD().key());
     String partitionPathField = context.getWriterContext().getProps().getString(DataSourceWriteOptions.PARTITIONPATH_FIELD().key());
     // todo: fix hard coded fields from configs.
     // read input and resolve insert, updates, etc.
     Dataset<Row> inputDf = session.read().format("avro").load(inputPath);
+    Dataset<Row> trimmedDf = inputDf;
+    if (!config.partitonsToSkipWithValidate().isEmpty()) {
+      trimmedDf = inputDf.filter("instr("+partitionPathField+", \'"+ config.partitonsToSkipWithValidate() +"\') != 1");
```

Comment:
This would work with the value i.e. '0' passed in the config. I think a more fool-proof way than substring check would be to use equals check with some FSUtil method to get relative partition path.

Reply:
then, we have to convert to timestamp based partition path and then do the filtering. here we are filtering on the input dataframe itself. So, we couldn't do much here.